### PR TITLE
First pass at TeamBroker

### PR DESCRIPTION
**Pull request by hardillb**
part of FlowFuse/flowfuse#1350

## Description

Add a Team Broker to the docker compose file.

## Related Issue(s)

FlowFuse/flowfuse#1350

## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Comment by hardillb** [comment]
Still need to do MQTT over TLS

**Review by ppawlowski — APPROVED** [review]

File: docker-compose.yml [hunk 1]
```diff
@@ -431,6 +431,8 @@ services:
         target: /etc/nginx/certs/${DOMAIN:?error}.crt
       - source: nginx_tls_key
         target: /etc/nginx/certs/${DOMAIN:?error}.key
+      - source: nginx_stream
+        target: /etc/nginx/toplevel.conf.d/mqtt.conf
```
Comment (ppawlowski) [inline docker-compose.yml:435]: How will nginx know about this custom `/etc/nginx/toplevel.conf.d` configuration directory?

Comment (hardillb) [reply docker-compose.yml:435]: https://github.com/nginx-proxy/nginx-proxy/tree/main/docs#tcp-and-udp-stream

File: docker-compose.yml [hunk 2]
```diff
@@ -150,6 +469,31 @@ services:
...
+      - "EMQX_DASHBOARD__DEFAULT_PASSWORD=topSecret"
+    configs:
+      - source: emqx
+        target: /opt/emqx/data/configs/cluster.hocon
+      - source: emqx-api
+        target: /mounted/config/api-keys
+    volumes:
+      - emqx:/opt/emqx/data
+    profiles:
+      - teamBroker
```
[the remark this answers is not in the document]
Comment (hardillb) [reply docker-compose.yml:496]: Need to decide here, without EE license old broker is fine.

But yeah, might be best to just install EMQX only, it should work, but it's over kill of CE small

**Comment by ppawlowski** [comment]
Overall, it looks good. Let's just make a call about the default broker, and I am okay with merging.

File: docker-compose.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (hardillb) [reply docker-compose.yml]: I've dropped the mosquitto broker

**Comment by hardillb** [comment]
@ppawlowski Can you give this one more quick look please

**Comment by hardillb** [comment]
We will leave quick start on the old broker this time.

File: docker-compose.yml [hunk 3]
```diff
@@ -30,30 +30,13 @@ configs:
         options:
           socket: /tmp/docker.sock
       broker:
-        url: mqtt://broker:1883
+        url: mqtt://teamBroker:1883
         public_url: ws${TLS_ENABLED:+s}://mqtt.${DOMAIN:?error}
+        teamBroker:
+          enabled: ${TEAM_BROKER_ENABLED}
```
Comment (ppawlowski) [inline docker-compose.yml:36]: If we use teamBroker as the default broker without a fallback to mosquito, do we still need this variable?

Comment (hardillb) [reply docker-compose.yml]: Removed

**Comment by hardillb** [comment]
I had to remove the MQTT WSS listener by default as nginx failed to start if there was no certs/key available

**Comment by ppawlowski** [comment]
I'm OK with merging this.

**Pull request merged**